core/src/test/scala/org/broadinstitute/dsde/rawls/bucketMigration/BucketMigrationServiceSpec.scala

```diff
@@ -491,6 +500,135 @@ class BucketMigrationServiceSpec extends AnyFlatSpec with TestDriverComponent {
+            _ <- dataAccess.workspaceQuery.createOrUpdate(nonUSWorkspace)
+            _ <- dataAccess.workspaceQuery.createOrUpdate(migratingWorkspace)
+            _ <- dataAccess.workspaceQuery.createOrUpdate(recentlyMigratedWorkspace)
+            _ <- dataAccess.workspaceQuery.createOrUpdate(oldSuccessfullyMigratedWorkspace)
+            _ <- dataAccess.workspaceQuery.createOrUpdate(oldUnsuccessfullyMigratedWorkspace)
+
+            _ <- dataAccess.multiregionalBucketMigrationQuery.schedule(migratingWorkspace, "US".some)
+            _ <- dataAccess.multiregionalBucketMigrationQuery.schedule(recentlyMigratedWorkspace, "US".some)
+            _ <- dataAccess.multiregionalBucketMigrationQuery.schedule(oldSuccessfullyMigratedWorkspace, "US".some)
+            _ <- dataAccess.multiregionalBucketMigrationQuery.schedule(oldUnsuccessfullyMigratedWorkspace, "US".some)
+
+            recentlyMigratedAttempt <- dataAccess.multiregionalBucketMigrationQuery
+              .getAttempt(recentlyMigratedWorkspace.workspaceIdAsUUID)
+              .value
+            oldSuccessfullyMigratedAttempt <- dataAccess.multiregionalBucketMigrationQuery
+              .getAttempt(oldSuccessfullyMigratedWorkspace.workspaceIdAsUUID)
+              .value
+            oldUnsuccessfullyMigratedAttempt <- dataAccess.multiregionalBucketMigrationQuery
+              .getAttempt(oldUnsuccessfullyMigratedWorkspace.workspaceIdAsUUID)
+              .value
+
+            _ <- dataAccess.multiregionalBucketMigrationQuery.update2(
+              recentlyMigratedAttempt.getOrElse(fail()).id,
+              multiregionalBucketMigrationQuery.finishedCol,
+              Timestamp.from(Instant.now().minusSeconds(86400)).some,
+              multiregionalBucketMigrationQuery.outcomeCol,
+              "Success".some
+            )
+            _ <- dataAccess.multiregionalBucketMigrationQuery.update2(
+              oldSuccessfullyMigratedAttempt.getOrElse(fail()).id,
+              multiregionalBucketMigrationQuery.finishedCol,
+              Timestamp.from(Instant.now().minusSeconds(86400 * 10)).some,
+              multiregionalBucketMigrationQuery.outcomeCol,
+              "Success".some
+            )
+            _ <- dataAccess.multiregionalBucketMigrationQuery.update3(
+              oldUnsuccessfullyMigratedAttempt.getOrElse(fail()).id,
+              multiregionalBucketMigrationQuery.finishedCol,
+              Timestamp.from(Instant.now().minusSeconds(86400 * 10)).some,
+              multiregionalBucketMigrationQuery.outcomeCol,
+              "Failure".some,
+              multiregionalBucketMigrationQuery.messageCol,
+              "reason the migration failed".some
+            )
+          } yield ()
+        },
+        Duration.Inf
+      )
+
+      val res = Await.result(bucketMigrationService.getEligibleOrMigratingWorkspaces, Duration.Inf)
+      res.map {
+        case (name, None) if name.equals(usWorkspace.toWorkspaceName.toString) =>
```

Comment:
🤔 It took me a bit to grok this assertion block. It might be clearer to express this as explicit gets against the map rather than a iteration over the map which is what this appears to be.